### PR TITLE
engine: add release notes for 20.10.23

### PR DESCRIPTION
**Pull request by thaJeztah**

### CODE REVIEW

File: engine/release-notes/index.md [hunk 1]
```diff
@@ -15,6 +15,38 @@ for Docker Engine.
...
+- Fix an issue where `docker build` would fail when using `--add-host=host.docker.internal:host-gateway`
+  with BuildKit enabled [moby/moby#44650](https://github.com/moby/moby/pull/44650){:target="_blank" rel="noopener"}.
+- Revert seccomp: block socket calls to `AF_VSOCK` in default profile [moby/moby#44712](https://github.com/moby/moby/pull/44712){:target="_blank" rel="noopener"}.
+  This change, while favorable from a security standpoint, caused a change
+  in behavior for some use-cases. As such, we are reverting it to ensure
+  stability and compatibility for the affected users.
+
+  However, users of `AF_VSOCK` in containers should recognize that this
+  (special) address family is not currently namespaced in any version of
+  the Linux kernel, and may result in unexpected behavior, like VMs
```
Comment (neersighted) [inline engine/release-notes/index.md:43]:
```suggestion
  the Linux kernel, and may result in unexpected behavior, like containers
```

I know I authored this, but it looks like I misspoke and created ambiguity here. Of course the VM can talk to its hypervisor if it listens on `AF_VSOCK`!

Comment (thaJeztah) [reply engine/release-notes/index.md]: Good catch; I _slightly_ rephrased to remove "20.10 branch" and similar, bit I think we overlooked that one in the PR, well 🤷‍♂️ 😂 

Either way: updated it here 🤗

**Review by tianon — DISMISSED** [review]

**Comment by thaJeztah** [comment]
release was tagged and published; https://github.com/moby/moby/releases/tag/v20.10.23

this should be ready to go (perhaps after @dvdksn did a review)

File: engine/release-notes/index.md [hunk 2]
```diff
@@ -15,6 +15,38 @@ for Docker Engine.
...
+- Update containerd (`containerd.io` package) to [v1.6.15](https://github.com/containerd/containerd/releases/tag/v1.6.15){:target="_blank" rel="noopener"}.
+- Update the package versioning format for `docker-compose-cli` to allow distro version updates [docker/docker-ce-packaging#822](https://github.com/docker/docker-ce-packaging/pull/822){:target="_blank" rel="noopener"}.
+- Update Go runtime to [1.18.10](https://go.dev/doc/devel/release#go1.18.minor){:target="_blank" rel="noopener"},
+
+### Bug fixes and enhancements
+
+- Fix an issue where `docker build` would fail when using `--add-host=host.docker.internal:host-gateway`
+  with BuildKit enabled [moby/moby#44650](https://github.com/moby/moby/pull/44650){:target="_blank" rel="noopener"}.
+- Revert seccomp: block socket calls to `AF_VSOCK` in default profile [moby/moby#44712](https://github.com/moby/moby/pull/44712){:target="_blank" rel="noopener"}.
+  This change, while favorable from a security standpoint, caused a change
```
Comment (dvdksn) [inline engine/release-notes/index.md:37]: Suggest to add a newline here to break the paragraph

```suggestion

  This change, while favorable from a security standpoint, caused a change
```

Comment (thaJeztah) [reply engine/release-notes/index.md:38]: @dvdksn updated; ptal 👍

**Review by dvdksn — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**